src/main.rs

```diff
@@ -45,11 +47,33 @@ struct Args {
     musl: bool,
     /// Don't actually build anything.
     dryrun: bool,
+    /// Where to obtain the source code for the package.
+    /// `crates-io`: Download the package from `crates.io`.
+    /// `project`: The package is in the current directory.
+    #[options(parse(try_from_str = "parse_source"))]
+    source: Option<Source>,
```

Comment:
I suspect that having this be a single flag may be clearer for users. So:
- `--crates`: your proposed `--source crates-io`
- (nothing): your proposed `--source project`

Otherwise they're burdened with needing to know what special string is necessary to pass to `--project`. A typo would cause an error, etc., and can't be auto-completed / suggested. Thoughts?

Reply:
The thinking on #28 was to potentially allow new future sources without needing a new flag, e.g `--source url` for a custom url. Do you provide autocompletions for cargo-aur currently? Can they be provided for switch commands like this?

Reply:
> Future-proofing eh. What other potential sources could there be?
> 
> > Do you provide autocompletions for cargo-aur currently?
> 
> Not currently. I think perhaps it would be possible to add the project/crates-io args to the completions as well, despite what I said, but let's not consider them for this PR.

url was the initial one I was thinking of - could open up to self-hosting or other git forges like Codeberg.

Reply:
How about this:
- `--crates`: your proposed `--source crates-io`
- (nothing): your proposed `--source project`
- (future) `--url`: the suggested future feature

And then make `--crates` and `--url` mutually exclusive at the flag-level.

Reply:
Do you think this is worth needing to add additional code to make the switch type flag mutually exclusive? Keep in mind `--source project` is not required currently, and user will get decent errors if the switch is entered incorrectly. Would it be better if the flag options were added to the help menu also?
_target/debug/cargo-aur: invalid argument to option `-s`: Invalid source type, expected `crates-io` or `project`._

Reply:
At the moment, since `--url` doesn't exist yet, it would just be `--crates` or "nothing". Let's go with this approach.

Reply:
Actioned.